src/log-query/src/log_query.rs

```diff
@@ -54,6 +54,50 @@ pub struct LogQuery {
     pub exprs: Vec<LogExpr>,
 }
 
+/// Nested filter structure that supports and/or relationships
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum Filters {
+    /// Single filter condition
+    Single(ColumnFilters),
+    /// Multiple filters with AND relationship
+    And(Vec<Filters>),
+    /// Multiple filters with OR relationship
+    Or(Vec<Filters>),
+    Not(Box<Filters>),
+}
+
+impl Default for Filters {
+    fn default() -> Self {
+        Filters::And(vec![])
+    }
+}
+
+impl From<ColumnFilters> for Filters {
+    fn from(filter: ColumnFilters) -> Self {
+        Filters::Single(filter)
+    }
+}
+
+impl From<Vec<ColumnFilters>> for Filters {
+    fn from(filters: Vec<ColumnFilters>) -> Self {
+        Filters::And(filters.into_iter().map(Filters::from).collect())
+    }
+}
```

Comment:
Should we provide this `From` implementation? I think  `Filters::and(other: Vec<ColumnFilters>)` and `Filters::or(other: Vec<ColumnFilters>)` are clear.